engine-sdk/src/lib.rs

```diff
@@ -41,6 +41,21 @@ pub fn log_utf8(bytes: &[u8]) {
         exports::log_utf8(bytes.len() as u64, bytes.as_ptr() as u64);
     }
 }
+/// Calls random seed produced by validators
+#[cfg(feature = "contract")]
+pub fn random_seed() -> H256 {
```

Comment:
I wonder if it makes sense to add this to `sdk::env::Env` instead of as a standalone sdk function.

This way we can have an implementation for it in the standalone engine too.

Reply:
I missed the Env trait, will implement it there instead. But I wonder what is the rule of thumb here. I was looking for the implementation of `sha256` and implementing `random_seed` next to it.

Reply:
The difference with `sha256` is that we have a correct implementation both with and without the `contract` feature because it is a pure mathematical function. That is the core principle: we must have the same behaviour for all functions in the wasm contract as in the standalone engine.